src/libxrpl/protocol/STAmount.cpp

```diff
@@ -385,53 +385,9 @@ operator+(STAmount const& v1, STAmount const& v2)
     if (v1.holds<MPTIssue>())
         return {v1.mAsset, v1.mpt().value() + v2.mpt().value()};
 
-    if (getSTNumberSwitchover())
-    {
-        auto x = v1;
-        x = v1.iou() + v2.iou();
-        return x;
-    }
-
-    int ov1 = v1.exponent(), ov2 = v2.exponent();
-    std::int64_t vv1 = static_cast<std::int64_t>(v1.mantissa());
-    std::int64_t vv2 = static_cast<std::int64_t>(v2.mantissa());
-
-    if (v1.negative())
-        vv1 = -vv1;
-
-    if (v2.negative())
-        vv2 = -vv2;
-
-    while (ov1 < ov2)
-    {
-        vv1 /= 10;
-        ++ov1;
-    }
-
-    while (ov2 < ov1)
-    {
-        vv2 /= 10;
-        ++ov2;
-    }
-
-    // This addition cannot overflow an std::int64_t. It can overflow an
-    // STAmount and the constructor will throw.
-
-    std::int64_t fv = vv1 + vv2;
-
-    if ((fv >= -10) && (fv <= 10))
-        return {v1.getFName(), v1.asset()};
-
-    if (fv >= 0)
-        return STAmount{
-            v1.getFName(),
-            v1.asset(),
-            static_cast<std::uint64_t>(fv),
-            ov1,
-            false};
-
-    return STAmount{
-        v1.getFName(), v1.asset(), static_cast<std::uint64_t>(-fv), ov1, true};
+    auto x = v1;
+    x = v1.iou() + v2.iou();
+    return x;
```

Comment:
```suggestion
    return v1.iou() + v2.iou();
```

Wouldn't this be the same? There must be something I'm missing here in terms of data typing or so.

Reply:
It's not the same.
When we do `auto x = v1`, we're actually doing `STAmount x = v1`, and then we call `STAmount::operator=(IOUAmount const& iou)` in the next statement to assign from an `IOUAmount`. On the other hand, there's not a constructor that converts from an `IOUAmount` to an `STAmount` for `return v1.iou() + v2.iou()` so we'll get a build error.

Reply:
But yeah, I think we should rewrite the code to make it a bit easier to understand. Potentially something like:
```c++
STAmount x{};
x = v1.iou() + v2.iou();
```

or we could make `STAmount` support constructing from an `IOUAmount`

Reply:
Yes please, the current approach is unclear. Either option works, although I think the first you proposed is easiest to add.